File: daily/面试题22. 链表中倒数第k个节点.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} k
 * @return {ListNode}
 */
var getKthFromEnd = function(head, k) {
    // var sum=0,p=head;
    // while(p)
    // {
    //     sum++;
    //     p=p.next;
    // }
    // var temp=sum-k;
    // while(temp--)
    // {
    //     head=head.next;
    // }
    // return head;

    var fast=head,slow=head,flag=0;
    while(fast)
    {
        if(flag>=k)
            slow=slow.next;
        fast=fast.next;
        flag++;
    }
    return slow;
};

File: daily/C++版本/面试题22. 链表中倒数第k个节点.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* getKthFromEnd(ListNode* head, int k) {
        // int sum=0;
        // ListNode *p=head;
        // while(p)
        // {
        //     p=p->next;
        //     sum++;
        // }
        // int temp=sum-k;
        // while(temp--)
        //     head=head->next;
        // return head;

        ListNode *fast=head,*slow=head;
        int flag=0;
        while(fast)
        {
            if(flag>=k)
                slow=slow->next;
            fast=fast->next;
            flag++;
        }
        return slow;
    }
};
